File: 20-valid-parentheses/valid-parentheses.ts
function isValid(s: string): boolean {
    const test: string[] = []
    const correct:{[key:string]:string}={
        ")":"(",
        "}":"{",
        "]":"["
    }

    for (let c of s){
        if(c==="(" || c==="[" || c==="{")
            test.push(c)
        else if  (c === ')' || c === '}' || c === ']') {
            if(test.length==0 || test.pop() !=correct[c])
            return false
        }
    }
    return test.length===0
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        test = []
        correct = {
            ')': '(',
            '}': '{',
            ']': '['
        }

        for char in s:
            if char in "({[":
                test.append(char)
            elif char in ")}]":
                if not test or test.pop() != correct[char]:
                    return False

        return len(test) == 0
